### PR TITLE
Endpoint Changes for Azure China Environment

### PR DESCRIPTION
**Pull request by v-rahkuma**
Endpoint Changes for Azure China Environment

### CODE REVIEW

**Review by asimks — APPROVED** [review]

File: Scripts/ScriptToSetupAzureADApplicationForAzTSUI.ps1 [hunk 1]
```diff
@@ -37,6 +37,8 @@ $AzureEnvironmentName = ""
 $AzureEnvironmentAppServiceURI = @{
     "AzureCloud" = "https://{0}.azurewebsites.net";
     "AzureGovernmentCloud" = "https://{0}.azurewebsites.us";
+    "AzureChinaCloud" = "https://{0}.azurewebsites.cn";
```
Comment (TarunKrShukla) [inline Scripts/ScriptToSetupAzureADApplicationForAzTSUI.ps1:40]: is this correct or this is correct value - https://{0}.chinacloudsites.cn

File: TemplateFiles/DeploymentFiles/AzTSDeploymentTemplate.json [hunk 2]
```diff
@@ -288,6 +288,14 @@
...
       "type": "string",
       "defaultValue": "00:09:00"
+    },
+    "AADClientAppDetailsInstance": {
+      "type": "string",
+      "defaultValue": "https://login.microsoftonline.com/"
+    },
+    "AzureEnvironmentPortalURI": {
+      "type": "string",
+      "defaultValue": "https://portal.azure.com/"
```
Comment (TarunKrShukla) [inline TemplateFiles/DeploymentFiles/AzTSDeploymentTemplate.json:298]: how we are changing it per environment?

File: TemplateFiles/DeploymentFiles/AzTSSetup.ps1 [hunk 3]
```diff
@@ -398,6 +410,11 @@ function Install-AzSKTenantSecuritySolution
                      $TemplateParameters.Add("FrontDoorEndpointSuffix", ".azurefd.us")
                      $TemplateParameters.Add("WebAppEndpointSuffix", ".azurewebsites.us")
                  }
+                 elseif($AzureEnvironmentName -eq 'AzureChinaCloud')
+                 {
+                     $TemplateParameters.Add("FrontDoorEndpointSuffix", ".azurefd.cn") 
+                     $TemplateParameters.Add("WebAppEndpointSuffix", ".azurewebsites.cn")
```
Comment (TarunKrShukla) [inline TemplateFiles/DeploymentFiles/AzTSSetup.ps1:416]: is this correct?

File: Scripts/ScriptToSetupAzureADApplicationForAzTSUI.ps1 (hunk 1 above)
Comment (v-rahkuma) [reply Scripts/ScriptToSetupAzureADApplicationForAzTSUI.ps1]: Updated the endpoint to https://{0}.chinacloudsites.cn

File: TemplateFiles/DeploymentFiles/AzTSDeploymentTemplate.json (hunk 2 above)
Comment (v-rahkuma) [reply TemplateFiles/DeploymentFiles/AzTSDeploymentTemplate.json:298]: This is the default value for the parameter in template and the actual value we are passing from AzTsSetup script as per environemnt like below:  $TemplateParameters.Add("AzureEnvironmentPortalURI", $AzureEnvironmentPortalURI.$AzureEnvironmentName).

I have tested the setup by removing the default value from template.

File: TemplateFiles/DeploymentFiles/AzTSSetup.ps1 (hunk 3 above)
Comment (v-rahkuma) [reply TemplateFiles/DeploymentFiles/AzTSSetup.ps1]: I have changed the value for WebAppEndpointSuffix to **.chinacloudsites.cn** .

**Pull request merged**